identity/src/commonTest/kotlin/com/android/identity/util/ByteArrayUtilTest.kt

```diff
@@ -0,0 +1,313 @@
+
+        byteArray.putUInt8(128)
+        assertEquals(128u, byteArray.getUInt8())
+
+        byteArray.putUInt8(UByte.MAX_VALUE.toInt())
+        assertEquals(UByte.MAX_VALUE, byteArray.getUInt8())
+    }
+
+    @Test
+    fun getUInt16_valid() {
+        val byteArray = ByteArray(2)
+        byteArray.putUInt16(UShort.MIN_VALUE.toInt())
+        assertEquals(UShort.MIN_VALUE, byteArray.getUInt16())
+
+        byteArray.putUInt16(32768)
+        assertEquals(32768u, byteArray.getUInt16())
+
+        byteArray.putUInt16(UShort.MAX_VALUE.toInt())
+        assertEquals(UShort.MAX_VALUE, byteArray.getUInt16())
+    }
+
+    @Test
+    fun getUInt32_valid() {
+        val byteArray = ByteArray(4)
+        byteArray.putUInt32(UInt.MIN_VALUE.toInt())
+        assertEquals(UInt.MIN_VALUE, byteArray.getUInt32())
+
+        byteArray.putUInt32(2147483647)
+        assertEquals(2147483647u, byteArray.getUInt32())
+
+        // The Int input should not be of UInt full range.
+        assertFailsWith<IllegalArgumentException> { byteArray.putUInt32(UInt.MAX_VALUE.toInt()) }
+    }
+
+    @Test
+    fun getUInt64_valid() {
+        val byteArray = ByteArray(8)
+        byteArray.putUInt64(ULong.MIN_VALUE.toLong())
+        assertEquals(ULong.MIN_VALUE, byteArray.getUInt64())
+
+        byteArray.putUInt64(4294967295)
+        assertEquals(4294967295u, byteArray.getUInt64())
+
+        // The Int input should not be of UInt full range.
+        assertFailsWith<IllegalArgumentException> { byteArray.putUInt64(ULong.MAX_VALUE.toLong()) }
+    }
+    @Test
+    fun getInt8_offset() {
+        val byteArray = ByteArray(3)
+        byteArray[1] = 100
+        assertEquals(100, byteArray.getInt8(1))
+    }
+
+    @Test
+    fun getInt16_offset() {
+        val byteArray = ByteArray(4)
+        byteArray.putInt16(256, 2)
+        assertEquals(256, byteArray.getInt16(2))
+    }
+
+    @Test
+    fun getInt32_offset() {
+        val byteArray = ByteArray(6)
+        byteArray.putInt32(1000000, 2)
+        assertEquals(1000000, byteArray.getInt32(2))
+    }
+
+    @Test
+    fun getInt64_offset() {
+        val byteArray = ByteArray(10)
+        byteArray.putInt64(10000000000, 2)
+        assertEquals(10000000000, byteArray.getInt64(2))
+    }
+
+    @Test
+    fun getUInt8_offset() {
+        val byteArray = ByteArray(3)
+        byteArray.putUInt8(200, 1)
+        assertEquals(200u, byteArray.getUInt8(1))
+    }
+
+    @Test
+    fun getUInt16_offset() {
+        val byteArray = ByteArray(4)
+        byteArray.putUInt16(60000, 2)
+        assertEquals(60000u, byteArray.getUInt16(2))
+    }
+
+    @Test
+    fun getUInt32_offset() {
+        val byteArray = ByteArray(6)
+        byteArray.putUInt32(3000000, 2)
+        assertEquals(3000000u, byteArray.getUInt32(2))
+    }
+
+    @Test
+    fun getUInt64_offset() {
+        val byteArray = ByteArray(10)
+        byteArray.putUInt64(10000000000, 2)
+        assertEquals(10000000000u, byteArray.getUInt64(2))
+    }
```

Comment:
None of these tests actually check that we're encoding things correctly, they just check our code is self-consistent. I would propose two things
- For the self-consistency checks (e.g. `assertEquals(value, get(put(value)))` use the word _roundtrip_ in the test name. This makes it easier for the reader to figure out what the test is doing (it's a commonly used term, I think).
- To check the encoding is correct, use toHex() on the byte array to check, e.g.

```
val foo = ByteArray(4)
foo.putInt32(0, 0x01020304)
assertEquals("01020304", foo.tohex())
foo.putInt32Le(0, 0x01020304)
assertEquals("04030201", foo.tohex())
```